### ISSUE TITLE
DaNN 在迭代过程中训练正确率和测试正确率一直都是0，且损失很大？

### ISSUE THREAD

**Issue by Mari987-g** [jindongwang/transferlearning#227]
王老师您好，
 我基于您DaNN 的算法我进行了修改，将二维的卷积改为一维，数据处理也进行了修改，然后将我自己的数据带进模型。在模型训练过程中我发现模型的训练正确率和测试正确率一直为0，想向您请教这个问题？

模型训练过程中出现的信息：
 20%|██        | 1/5 [00:12<00:51, 12.83s/it]Test: total loss: 15331037.000000, correct: [0/3000], testing accuracy: 0.0000%
 20%|██        | 1/5 [00:25<00:51, 12.83s/it]Epoch: [2/5], training loss: 0.107117, correct: [0/3000], training accuracy: 0.0000%
 40%|████      | 2/5 [00:26<00:40, 13.37s/it]Test: total loss: 22273666.000000, correct: [0/3000], testing accuracy: 0.0000%
 40%|████      | 2/5 [00:40<00:40, 13.37s/it]Epoch: [3/5], training loss: 0.070205, correct: [0/3000], training accuracy: 0.0000%
 60%|██████    | 3/5 [00:41<00:28, 14.07s/it]Test: total loss: 30196982.000000, correct: [0/3000], testing accuracy: 0.0000%
 60%|██████    | 3/5 [00:56<00:28, 14.07s/it]Epoch: [4/5], training loss: 0.056889, correct: [0/3000], training accuracy: 0.0000%
 80%|████████  | 4/5 [00:57<00:14, 14.86s/it]Test: total loss: 33327742.000000, correct: [0/3000], testing accuracy: 0.0000%
 80%|████████  | 4/5 [01:13<00:14, 14.86s/it]Epoch: [5/5], training loss: 0.064430, correct: [0/3000], training accuracy: 0.0000%
100%|██████████| 5/5 [01:14<00:00, 14.95s/it]Test: total loss: 29025662.000000, correct: [0/3000], testing accuracy: 0.0000%

我的模型：
class DaNN(nn.Module):
    def __init__(self, n_class=8):
        super(DaNN, self).__init__()
        self.conv1 = nn.Conv1d(in_channels=1,out_channels=64,kernel_size=3)
        self.conv2 = nn.Conv1d(in_channels=64,out_channels=128,kernel_size=3)
        self.conv3 = nn.Conv1d(in_channels=128,out_channels=256,kernel_size=3)
        self.fc1 = nn.Linear(256*58,516)
        self.dropout = nn.Dropout(p=0.5)
        self.relu = nn.ReLU()
        self.layer_out = nn.Linear(516, n_class)

    def forward(self, src, tar):
        x_src = F.relu(self.conv1(src))
        x_tar = F.relu(self.conv1(tar))
        x_src = F.relu(self.conv2(x_src))
        x_tar = F.relu(self.conv2(x_tar))
        x_src = F.relu(self.conv3(x_src))
        x_tar = F.relu(self.conv3(x_tar))
        x_src = x_src.view(-1,256*58)
        x_tar = x_tar.view(-1,256*58)
        x_src = F.relu(self.fc1(x_src))
        x_tar = F.relu(self.fc1(x_tar))
        x_src = self.dropout(x_src)
        x_tar = self.dropout(x_tar)
        x_src_mmd = self.relu(x_src)
        x_tar_mmd = self.relu(x_tar)
        y_src = self.layer_out(x_src_mmd)
        return y_src, x_src_mmd, x_tar_mmd

数据加载部分

def load_data(BATCH_SIZE):
    
    df=pd.read_csv('E:\\A.csv',header = None,engine='python')
    dt=pd.read_csv('E:\\D.csv',header = None, engine='python')
    X_train0=df.iloc[:,1:65]
    X_test0=dt.iloc[:,1:65]
    y_sur=df.iloc[:,0]
    y_tar=dt.iloc[:,0]

    mms=MinMaxScaler()
    train_std=mms.fit_transform(X_train0)
    test_std=mms.transform(X_test0)
    X_train=pd.DataFrame(train_std,index=X_train0.index.values,columns=X_train0.columns)
    X_test=pd.DataFrame(test_std,index=X_test0.index.values,columns=X_test0.columns)
    X_sur = np.array(X_train).astype('float32')
    X_tar = np.array(X_test).astype('float32')
    y_sur = np.array(y_sur).astype('int64')
    y_tar = np.array(y_tar).astype('int64')
    
    X_sur, X_tar, y_sur, y_tar = map(torch.tensor, (X_sur, X_tar, y_sur, y_tar))
    X_sur = X_sur.reshape(X_sur.shape[0],1,X_sur.shape[1])
    X_tar = X_tar.reshape(X_tar.shape[0],1,X_tar.shape[1])
    print(X_sur.shape, X_tar.shape)
    
    y_sur = y_sur.reshape(y_sur.shape[0],)
    y_tar = y_tar.reshape(y_tar.shape[0],)
    y_sur = torch.tensor(y_sur, dtype=torch.long)
    y_tar = torch.tensor(y_tar, dtype=torch.long)
    
    sur_dataset = Data.TensorDataset(X_sur, y_sur)
    tar_dataset = Data.TensorDataset(X_tar, y_tar)
    loader_sur = Data.DataLoader(dataset=sur_dataset, batch_size= BATCH_SIZE ,shuffle=True, num_workers=0)
    loader_tar = Data.DataLoader(dataset=tar_dataset, batch_size= BATCH_SIZE ,shuffle=True, num_workers=0)
    
    return loader_sur, loader_tar

模型其余部分基本没变，

请问我这是哪里出现了问题？

**Comment by Mari987-g** [comment]
王老师，我又重新建立一个和这个模型相同结构的一维CNN模型，并采用这个数据加载部分将数据输入进模型，对模型进行训练，最后得到很好的结果，这是不是证明数据加载部分没有问题呀？

**Comment by jindongwang** [comment]
那说明没问题，是你的模型有问题。现在问题是解决了是吧？

**Comment by Mari987-g** [comment]
王老师，问题还没有解决。我将新建立的CNN模型带进DaNN模型中，就出现这个问题，现在还不知从何下手解决这个问题。

**Comment by jindongwang** [comment]
这个和DaNN无关。你用DDC试一下，把你的换了DDC里面的backbone就可以了，其他都不用改

**Comment by jindongwang** [comment]
你的conv 1--3 和fc1都是Feature，最后一层是分类器，对应换一下就行了，非常简单
